# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/debug/bug1602392.js
# Commit: 19f956b37c6c
# Full Hash: 19f956b37c6cb832b41b8d1ffb8648f38336f4dd
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2019-12-19 21:52:02
# Description:
#   Bug 1602392 - Change for-of bytecode to not have an extra value on the stack across the loop edge. r=arai
#   
#   This fixes a bug with HasLiveStackValueAtDepth because now all stack values that
#   are part of the loop (2 values instead of 3 values) are loop-invariant.
#   
# ==============================================================================

diff -r f1547816cf32 -r 19f956b37c6c js/src/jit-test/tests/debug/bug1602392.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/debug/bug1602392.js	Thu Dec 19 09:06:01 2019 +0000
@@ -0,0 +1,19 @@
+// |jit-test| error:ReferenceError: iter is not defined
+var g = newGlobal({newCompartment: true});
+g.parent = this;
+g.eval("new Debugger(parent).onExceptionUnwind = function () {};");
+function* f1() {
+    for (const x of iter) {
+        yield x;
+    }
+}
+function f2() {
+    for (var i of [1, 2, 3]) {
+        for (var j of [4, 5, 6]) {
+            for (const k of f1()) {
+                break;
+            }
+        }
+    }
+}
+f2();
